qlib/TableMapper.qm

```diff
@@ -597,32 +603,58 @@ table_mapper.setRowCode(rowcode);
-on_success table_mapper.commit();
-on_error table_mapper.rollback();
+# Example 1:
 {
     on_success table_mapper.flush();
     on_error table_mapper.discard();
 
-    map table_mapper.queueData($1), data.iterator();
+    while (*hash h = stmt.fetchColumns(1000)) {
+        table_mapper.queueData(h);
+    }
+}
+
+# Example 2:
+{
+    const Map1 = (
+        "num": ("a"),
+        "str": ("b"),
+    );
+
+    Table table(someDataSource, "some_table");
+    InboundTableMapper table_mapper(table, DataMap. ("insert_block" : 3));
+
+    on_error table_mapper.discard();
+
+    list data = (("a" : 1, "b" : "bar"), ("a" : 2, "b" : "foo"));
+
+    list mapped_data = (map table_mapper.queueData($1), data.iterator()) ?? ();
+    # mapped_data = () - no insertion done yet since to insert_block=3
+
+    list flushed_data = table_mapper.flush() ?? ();
+
+    mapped_data += flushed_data;
+    # mapped_data = ("num" : (1,2), "str" : ("bar","foo"));
+    # the table is updated too
 }
             @endcode
 
             Data is only inserted if the block buffer size reaches the limit defined by the \c "insert_block" option, in which case this method returns all the data inserted.  In case the mapped data is only inserted into the cache, no value is returned.
 
             @param rec the input record or record set in case a hash of lists is passed
-            @param crec an optional simple hash of data to be added to each row
+            @param crec an optional simple hash of data to be added to each input row before mapping
 
-            @return if batch data was inserted then a hash (columns) of lists (row data) of all data inserted and potentially returned (in case of sequences) from the database server is returned; if constant mappings are used with batch data, then they are returned as single values assigned to the hash keys
+            @return if batch data was inserted then a hash (columns) of lists (row data) of all data inserted and potentially returned (in case of sequences) from the database server is returned; if constant mappings are used with batch data, then they are returned as single values assigned to the hash keys; always returns a bulk (hash of lists)
 
             @note
             - make sure to call flush() before committing the transaction or discard() before rolling back the transaction or destroying the object when using this method
             - flush() or discard() needs to be executed for each mapper used in the block when using multiple mappers whereas the DB transaction needs to be committed or rolled back once per datasource
             - this method and batched inserts in general cannot be used when the \c "unstable_input" option is given in the constructor
-            - if the \c "insert_block" option is set to 1, then this method simply calls insertRow()
+            - if the \c "insert_block" option is set to 1, then this method simply calls insertRow(); even in this case a bulk is returned, though
```

Comment:
I don't understand the new comment - I think a noun is missing after the adjective `bulk` - maybe it should read `however please note that in this case the return value is a hash of single value lists corresponding to a batch data insert`